Firestore/Swift/Source/Codable/third_party/FirestoreDecoder.swift

```diff
@@ -323,6 +330,17 @@ private struct _FirestoreKeyedDecodingContainer<K: CodingKey>: KeyedDecodingCont
   }
 
   public func decode<T: Decodable>(_: T.Type, forKey key: Key) throws -> T {
+    if T.self == AutoPopulatedDocumentId.self {
+      if contains(key) {
+        throw FirestoreDecodingError.fieldNameConfict("Field name conflict:" +
```

Comment:
This error message should name the offending document (see, e.g. [Android](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java#L814)).

Also, since we have the coding path, use that to name the field that's problematic?